lib/jsdom/browser/new-resource-loader.js

```diff
@@ -0,0 +1,131 @@
+const DEFAULT_USER_AGENT = `Mozilla/5.0 (${process.platform}) AppleWebKit/537.36 (KHTML, like Gecko) ` +
+  `jsdom/${packageVersion}`;
+const IS_BROWSER = Object.prototype.toString.call(process) !== "[object process]";
+
+class ResourceLoader {
+  constructor() {
+    this._defaultOptions = {
+      agentOptions: {
+        keepAlive: true,
+        keepAliveMsecs: 115 * 1000
+      },
+      pool: { maxSockets: 6 },
+      strictSSL: true
+    };
+  }
+
+  _readFile(filePath) {
+    let readableStream;
+    let abort; // Native Promises doesn't have an "abort" method.
+
+    /*
+     * Creating a promise for two reason:
+     *   1. fetch always return a promise.
+     *   2. We need to add an abort handler.
+    */
+    const promise = new Promise((resolve, reject) => {
+      readableStream = fs.createReadStream(filePath);
+      let data = Buffer.alloc(0);
+
+      abort = reject;
+
+      readableStream.on("error", reject);
+
+      readableStream.on("data", chunk => {
+        data = Buffer.concat([data, chunk]);
+      });
+
+      readableStream.on("end", () => {
+        resolve(data);
+      });
+    });
+
+    promise.abort = () => {
+      readableStream.destroy();
+      const error = new Error("request canceled by user");
+      error.isAbortError = true;
+      abort(error);
+    };
+
+    return promise;
+  }
+
+  /**
+   * Returns the options to make the request
+   * @param {object} options - Custom options for the request.
+   */
+  _getRequestOptions(options = {}) {
+    const rqOptions = options.requestOptions;
+    const requestOptions = {
+      agent: rqOptions.agent,
+      agentOptions: rqOptions.agentOptions ?
+        Object.assign({}, this._defaultOptions.agentOptions, rqOptions.agentOptions) :
+        this._defaultOptions.agentOptions,
+      agentClass: rqOptions.agentClass,
+      encoding: null,
+      gzip: true,
+      jar: rqOptions.jar,
+      pool: rqOptions.pool ? Object.assign({}, this._defaultOptions.pool, rqOptions.pool) : this._defaultOptions.pool,
+      // If strictSSL is not defined, then strictSSL will be true.
+      strictSSL: rqOptions.strictSSL !== false,
+      headers: {
+        "User-Agent": rqOptions.userAgent || DEFAULT_USER_AGENT,
+        "Accept-Language": "en",
+        Accept: rqOptions.accept || "*/*"
+      }
+    };
+
+    if (rqOptions.referrer && !IS_BROWSER) {
+      requestOptions.headers.referer = options.referrer;
+    }
+    if (rqOptions.proxy) {
+      requestOptions.proxy = rqOptions.proxy;
+    }
+
+    return requestOptions;
+  }
+
+  /**
+   * Fetch a resource.
+   * @param {string} urlString - Absolute url (data URLs are supported).
+   * @param {object} options - Options to configure fetch.
+   */
+  fetch(urlString, options = {}) {
+    if (!urlString) {
+      return Promise.reject(new Error("Invalid url string"));
+    }
+
+    let url = parseURL(urlString);
+
+    // Question: Can we assume this?
```

Comment:
This seems wrong. Why wouldn't the URL string be a `file://` URL?

I'd rather this section omit the empty-string check, and just always reject if parsing fails.

Reply:
I will take a look to all of this, because when I change to the `parseURL` I had a lot of problems with tests. I'm going to try all the changes related with this and try to explain the problems I have.

Reply:
Ok, that was because there are a couple of tests that use the `script` property in the old API. and they don't use `file://`

I will fail and modify those tests